<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>919</width>
    <height>461</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frequency Analysis Test</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="leftMargin">
     <number>8</number>
    </property>
    <property name="topMargin">
     <number>8</number>
    </property>
    <property name="rightMargin">
     <number>8</number>
    </property>
    <property name="bottomMargin">
     <number>8</number>
    </property>
    <property name="horizontalSpacing">
     <number>0</number>
    </property>
    <item row="0" column="0" rowspan="3">
     <widget class="QListWidget" name="listWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>175</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>175</width>
        <height>16777215</height>
       </size>
      </property>
      <item>
       <property name="text">
        <string>Multiple Frequency Test</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Sine Sweep</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Settings</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>About</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>26</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">border:none</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>12</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QCheckBox" name="checkBox_2">
         <property name="text">
          <string>Sound Output</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_9">
           <property name="font">
            <font>
             <pointsize>22</pointsize>
            </font>
           </property>
           <property name="text">
            <string>ðŸ”Š</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSlider_2">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>26</height>
            </size>
           </property>
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="page_1">
       <widget class="QListWidget" name="waveListWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>21</y>
          <width>391</width>
          <height>199</height>
         </rect>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>259</y>
          <width>68</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton">
        <property name="geometry">
         <rect>
          <x>420</x>
          <y>253</y>
          <width>301</width>
          <height>32</height>
         </rect>
        </property>
        <property name="text">
         <string>Start Analysis</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_3">
        <property name="geometry">
         <rect>
          <x>320</x>
          <y>231</y>
          <width>81</width>
          <height>54</height>
         </rect>
        </property>
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
       <widget class="QComboBox" name="comboBox_4">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>231</y>
          <width>71</width>
          <height>32</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Hz</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>kHz</string>
         </property>
        </item>
       </widget>
       <widget class="QLineEdit" name="lineEdit_2">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>235</y>
          <width>69</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QComboBox" name="comboBox_5">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>256</y>
          <width>71</width>
          <height>32</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>dB</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>420</x>
          <y>2</y>
          <width>181</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Spectrum Analysis Table</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButton_2">
        <property name="geometry">
         <rect>
          <x>230</x>
          <y>231</y>
          <width>81</width>
          <height>54</height>
         </rect>
        </property>
        <property name="text">
         <string>Add</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>-1</y>
          <width>121</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Sine Wave Table</string>
        </property>
       </widget>
       <widget class="QTableWidget" name="tableWidget">
        <property name="geometry">
         <rect>
          <x>420</x>
          <y>22</y>
          <width>301</width>
          <height>221</height>
         </rect>
        </property>
        <column>
         <property name="text">
          <string>Frequency</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Amplitude</string>
         </property>
        </column>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>237</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Frequency:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>260</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Amplitude:</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_2">
       <widget class="QPushButton" name="pushButton_4">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>251</y>
          <width>301</width>
          <height>32</height>
         </rect>
        </property>
        <property name="text">
         <string>Start Analysis</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>0</y>
          <width>181</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Spectrum Analysis Table</string>
        </property>
       </widget>
       <widget class="QTableWidget" name="tableWidget_2">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>20</y>
          <width>301</width>
          <height>221</height>
         </rect>
        </property>
        <column>
         <property name="text">
          <string>Frequency</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Amplitude</string>
         </property>
        </column>
       </widget>
       <widget class="QSlider" name="frequencySlider">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>381</width>
          <height>25</height>
         </rect>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>0</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Frequency:</string>
        </property>
       </widget>
       <widget class="QLabel" name="frequencyLabel">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>0</y>
          <width>61</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>0.5</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="frequencyUnitLabel">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>0</y>
          <width>58</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Hz</string>
        </property>
       </widget>
       <widget class="QSlider" name="horizontalSlider_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>100</y>
          <width>381</width>
          <height>25</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_13">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Amplitude:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_14">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>70</y>
          <width>61</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_15">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>70</y>
          <width>58</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>dB</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_3">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="frequencyAnalysisSettingsTab">
           <attribute name="title">
            <string>Frequency Analysis Settings</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="1" column="1">
               <widget class="QLabel" name="bandDesignatorLabel">
                <property name="text">
                 <string>1</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Band Designator (b)</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="bandAmountLabel">
                <property name="text">
                 <string>10</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Analysis Mode</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="analysisModeComboBox">
                <item>
                 <property name="text">
                  <string>Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/2 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/3 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/4 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/6 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/8 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/12 Octave Bands</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1/24 Octave Bands</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>Band Amount</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QTableWidget" name="octaveBandsTable">
              <property name="sizeAdjustPolicy">
               <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
              </property>
              <attribute name="horizontalHeaderCascadingSectionResizes">
               <bool>true</bool>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Index X</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Nominal Mid-Band Freq</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Exact Mid-Band Freq</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Beginning Freq</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Ending Freq</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Bandwidth</string>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="soundSettingsTab">
           <attribute name="title">
            <string>Sound Settings</string>
           </attribute>
           <widget class="QWidget" name="gridLayoutWidget">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>10</y>
              <width>551</width>
              <height>131</height>
             </rect>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <property name="verticalSpacing">
              <number>-1</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Sound Device</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_3">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Sampling Frequency</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Bit Rate</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="comboBoxSoundDevices"/>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="bitDepths">
               <item>
                <property name="text">
                 <string>8 Bit</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>16 Bit</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>24 Bit</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="samplingFrequencies">
               <item>
                <property name="text">
                 <string>11 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>22 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>24 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>44.1 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>48 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>96 kHz</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>192 kHz</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_4">
       <widget class="QLabel" name="label_17">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>-3</y>
          <width>171</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Frequency Analysis Test</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_18">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>27</y>
          <width>200</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Copyright Volkan Orhan - 2025</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_19">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>57</y>
          <width>161</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Andromeda Signal Lab</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
